### PR TITLE
ns ttl update retry.

### PR DESCRIPTION
**Pull request by droslean**
Solves errors like: 
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/20616/pull-ci-origin-e2e-gcp/2300/build-log.txt

### CODE REVIEW

**Comment by bbguimaraes** [comment]
And…

/approve

**Comment by bbguimaraes** [comment]
=(

File: cmd/ci-operator/main.go [hunk 1]
```diff
@@ -581,16 +581,20 @@ func (o *options) initializeNamespace() error {
...
+		for attempts := 0; attempts < 5; attempts++ {
+			ns, err := client.Namespaces().Get(o.namespace, meta.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("could not retrieve namespace: %v", err)
+			}
 
-		update := ns.DeepCopy()
-		for key, value := range updates {
-			update.ObjectMeta.Annotations[key] = value
+			if _, err := client.Namespaces().Update(ns); err != nil && kerrors.IsConflict(err) {
```
Comment (smarterclayton) [inline cmd/ci-operator/main.go]: you need to return any non-conflict error.

Please use client.RetryOnConflict in the client-go library for this

File: cmd/ci-operator/main.go [hunk 2]
```diff
@@ -581,16 +581,20 @@ func (o *options) initializeNamespace() error {
 	}
 
 	if len(updates) > 0 {
-		ns, err := client.Namespaces().Get(o.namespace, meta.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("could not retrieve namespace: %v", err)
-		}
+		for attempts := 0; attempts < 5; attempts++ {
+			ns, err := client.Namespaces().Get(o.namespace, meta.GetOptions{})
```
Comment (smarterclayton) [inline cmd/ci-operator/main.go]: You have to apply your modification to the object each time you retrieve it

File: cmd/ci-operator/main.go [hunk 3]
```diff
@@ -581,17 +582,21 @@ func (o *options) initializeNamespace() error {
 	}
 
 	if len(updates) > 0 {
-		ns, err := client.Namespaces().Get(o.namespace, meta.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("could not retrieve namespace: %v", err)
-		}
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```
Comment (smarterclayton) [inline cmd/ci-operator/main.go]: nit: by convention we always reuse error, so make this `if err := ...; err != nil {`

**Pull request merged**